### ISSUE TITLE
Two classes of widgets : rendered and headless

### ISSUE THREAD

**Issue by bobylito** [algolia/instantsearch.js#44]
Maybe the name are not great but the idea is the following : when using our widget we might feel that we want to let it generates markup (because we don't have a component for that or we don't have a search page yet) or not (BOOTSTRAP EVERYTHING!). And some components make a lot of sense in those two configurations, such as : 
 - searchBox (onchange event will give us the new value)
 - indexSelector (onchange)

That's all that come in mind for now. But I think this list could get longer as we go further and we have more components.

Are you ok with having those two kind of widget living as widgets but under different sub namespace?

**Comment by vvo** [comment]
I do think it would make our API more complex (from a user POV) to have something like:

```js
widgets.headLess.searchBox
```

or

```js
instantsearch(appId, apiKey, {searchBox:}
```

(also this second proposal from @redox does not allow the user to put multiple searchBox, something people will do at some point)

Versus just having:

```js
widgets.searchBox
```

Why would we distinguish the searchBox from other widgets only because we do not insert markup in it?. What level of information it brings to the API user?

Even if we do not insert markup we are going to do a LOT of DOM: bind event handlers, change placeholder, change the value of the input.

As a user I know that using instantsearch.js will do DOM manipulation. If you warn me that "The searchBox widgets works by enhancing an <input> you provide" Then I am allright, it's all I need.

**To summarize**: `searchBox` IS a widget, it's just an empty element that we do DOM manipulation on it, no need to complexify the API for this implementation detail that we can express in plain words in the documentation.

Trying to put a one line documentation detail into our API design will result in more complexity and incomprehension from users.

**Comment by pixelastic** [comment]
Agreed with @vvo, I would not split widgets so early in our development phase. We know today that those widgets will not need to generate markup, but maybe they will in the future with feature requests.

**Comment by vvo** [comment]
I do think we can close this, feel free to reopen if someone still wants and needs to split widgets namespaces.
